nexus/src/app/mod.rs

```diff
@@ -245,6 +297,56 @@ impl Nexus {
         }
     }
 
+    // Called to hand off management of external servers to Nexus.
+    pub(crate) async fn set_servers(
+        &self,
+        external_servers: ExternalServers,
+        internal_server: DropshotServer,
+    ) {
+        // If any servers already exist, close them.
+        let _ = self.close_servers().await;
+
+        // Insert the new servers.
+        *self.external_servers.lock().await = external_servers;
+        self.internal_server.lock().await.replace(internal_server);
+    }
+
+    pub async fn close_servers(&self) -> Result<(), String> {
+        let mut external_servers = self.external_servers.lock().await;
+        if let Some(server) = external_servers.http.take() {
+            server.close().await?;
+        }
+        if let Some(server) = external_servers.https.take() {
+            server.close().await?;
+        }
+        if let Some(server) = self.internal_server.lock().await.take() {
+            server.close().await?;
+        }
```

Comment:
Can/should these be parallelized?

Reply:
They can be, though I'm not sure they should be - this operation isn't very common.